python/pyarrow/io.pxi

```diff
@@ -1368,6 +1456,26 @@ cdef class BufferReader(NativeFile):
     Parameters
     ----------
     obj : Python bytes or pyarrow.Buffer
+
+    Examples
+    --------
+
+    >>> import pyarrow as pa
+    >>> data = b'reader data'
+    >>> f = pa.BufferReader(data)
```

Comment:
We should maybe also reference `pa.input_stream()` as the (preferred?) way to create BufferReader objects?

Reply:
(and same for some of the others)